### ISSUE TITLE
New Data source - aws_transfer_server

### ISSUE THREAD

**Issue by atsushi-ishibashi** [terraform-providers/terraform-provider-aws#7976]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Data source for transfer server

### New or Affected Resource(s)

* aws_transfer_server

### Potential Terraform Configuration

```hcl
data "aws_transfer_server" "foo" {
  server_id = "s-1234567"
}
```

### References

https://docs.aws.amazon.com/transfer/latest/userguide/API_DescribeServer.html

**Comment by bflad** [comment]
A new `aws_transfer_server` data source has been merged and will release with version 2.3.0 of the Terraform AWS Provider, likely middle of this week.

**Comment by bflad** [comment]
This has been released in [version 2.3.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#230-march-21-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
